codes/models/modules/architectures/DVDNet_arch.py

```diff
@@ -0,0 +1,68 @@
+import torch
+import torch.nn as nn
+
+def vertcal_upscale(x, upfield=True):
+
+    n, c, h, w = x.shape
+    h *= 2
+    t = x
+    if upfield:
+        out = torch.cat([t, torch.zeros_like(t)], 3)
+    else:
+        out = torch.cat([torch.zeros_like(t), t], 3)
+    out = torch.reshape(out, (n, c, h, w))
+    return out
+
+
+def replace_field(x, input_image, upfield=True):
+
+    upper_input = input_image[:, :, 0::2, :]
+    lower_input = input_image[:, :, 1::2, :]
+    # print(upper_input.shape, lower_input.shape)
+
+    if upfield:
+        # print(upper_input.shape, x.shape)
+        x = vertcal_upscale(x, upfield=False)
+        upper_input = vertcal_upscale(upper_input, upfield=True)
+        # print(upper_input.shape, x.shape)
+        out = x + upper_input
+    else:
+        x = vertcal_upscale(x, upfield=True)
+        lower_input = vertcal_upscale(lower_input, upfield=False)
+        out = x + lower_input
+
+    return out
+
+
+class DVDNet(nn.Module):
+    '''
+    Real-time Deep Video Deinterlacing: https://arxiv.org/pdf/1708.00187.pdf
+    Original TF code: https://github.com/lszhuhaichao/Deep-Video-Deinterlacing
+    '''
+    def __init__(self, in_nc=3, out_nc=3, nf=64):
+        super(DVDNet, self).__init__()
+        conv_fea_1 = nn.Sequential(nn.Conv2d(in_nc, nf, 3), nn.ReLU())
+        conv_fea_2 = nn.Sequential(nn.Conv2d(nf, nf, 3), nn.ReLU())
+        conv_fea_3 = nn.Conv2d(nf, nf//2, 1)
+        h = nn.Sequential(conv_fea_1, conv_fea_2, conv_fea_3)
+
+        conv_branch_top = nn.Conv2d(nf//2, nf//2, 3)
+        conv_branch_bottom = nn.Conv2d(nf//2, nf//2, 3)
+
+        final_branch_top = nn.Conv2d(
+            nf//2, out_nc, 3, stride=(2, 1), padding=4, padding_mode='replicate')

```

Comment:
Both H and W dimensions need padding? Else, use correct (H, W) tuple

Reply:
both need padding of 4 for some reason